TypeError: Invalid first argument to register()

* Description of issue
Using Python 3.7.0,  Can not import niswitch (0.9.0 and 1.0.0)

    ```
import niswitch
C:\Program Files (x86)\Python37-32\lib\site-packages\niswitch\init.py:7: in <module>
    from niswitch.session import Session  # noqa: F401
C:\Program Files (x86)\Python37-32\lib\site-packages\niswitch\session.py:7: in <module>
    import niswitch._attributes as _attributes
C:\Program Files (x86)\Python37-32\lib\site-packages\niswitch\_attributes.py:3: in <module>
    import niswitch._converters as _converters
C:\Program Files (x86)\Python37-32\lib\site-packages\niswitch\_converters.py:54: in <module>
    @_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
C:\Program Files (x86)\Python37-32\lib\functools.py:801: in register
    f"Invalid first argument to `register()`: {cls!r}. "
E   TypeError: Invalid first argument to `register()`: (<class 'str'>,). Use either `@register(some_class)` or plain `@register` on an annotated function.
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!
    ```

* Steps to reproduce issue

    ``` python
    import niswitch
    ```

This may be of interest:
https://github.com/python/cpython/commit/e56975351bc2c574c728f738e88effba7116279f

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,7 +25,6 @@ jobs:
         sudo: true
 matrix:
     allow_failures:
-        - python: 3.7      # remove once #895 is fixed
         - python: 3.8-dev
         - python: nightly
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ All notable changes to this project will be documented in this file.
 ## [Unreleased]
 * ### ALL
     * #### Added
+        * Support for Python 3.7
     * #### Changed
     * #### Removed
 * ### NI-DMM

--- a/build/templates/_converters.py.mako
+++ b/build/templates/_converters.py.mako
@@ -55,7 +55,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/build/templates/setup.py.mako
+++ b/build/templates/setup.py.mako
@@ -64,6 +64,7 @@ setup(
         "Programming Language :: Python :: 3.4",
         "Programming Language :: Python :: 3.5",
         "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
         "Topic :: System :: Hardware :: Hardware Drivers"

--- a/generated/nidcpower/_converters.py
+++ b/generated/nidcpower/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/generated/nidmm/_converters.py
+++ b/generated/nidmm/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/generated/nifake/_converters.py
+++ b/generated/nifake/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/generated/nifgen/_converters.py
+++ b/generated/nifgen/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/generated/nimodinst/_converters.py
+++ b/generated/nimodinst/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/generated/niscope/_converters.py
+++ b/generated/niscope/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/generated/niswitch/_converters.py
+++ b/generated/niswitch/_converters.py
@@ -51,7 +51,6 @@ def _(repeated_capability, prefix):
 
 
 # This parsing function duplicate the parsing in the driver, so if changes to the allowed format are made there, they will need to be replicated here.
-@_convert_repeated_capabilities.register(six.string_types)  # noqa: F811
 @_convert_repeated_capabilities.register(six.text_type)  # noqa: F811
 @_convert_repeated_capabilities.register(str)  # noqa: F811
 def _(repeated_capability, prefix):

--- a/tox.ini
+++ b/tox.ini
@@ -5,6 +5,7 @@
 [tox]
 envlist = build_test, codegen, py{27,34,35,36,37,38,py,py3}-test, flake8, docs, pkg-{nidmm,nidcpower,nifgen,niscope,niswitch,nimodinst}
 skip_missing_interpreters=True
+ignore_basepython_conflict=True
 skipsdist = true
 toxworkdir = .tox/{env:BITNESS:64}
 

